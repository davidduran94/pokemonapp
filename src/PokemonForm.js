import React from "react";

function PokemonForm(props){
    const style = {

    }

    let searchPokemon = (evt) => {
        evt.preventDefault();
        let name_pokemon = evt.target.pokemonName.value;
        props.onNewPokemonSearch(name_pokemon);
    }

    return (
        <form onSubmit={searchPokemon}>
            <label></label>
            <input type="text" name="pokemonName">
            </input>
            <input type="submit" value="Add Pokemon" />
        </form>
    )
}

export default PokemonForm;