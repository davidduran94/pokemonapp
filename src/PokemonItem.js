import "./styles/pokemonItem.css"
import React from "react";

function PokemonItem(props){
    return (
        <div className="poke">
            <img className="poke_img" src={props.data.img} alt="pokemon"/>
            <p className="poke_name">{props.data.name}</p>
            <p className="poke_id">{props.data.id}</p>
        </div>
    )
}

export default PokemonItem;