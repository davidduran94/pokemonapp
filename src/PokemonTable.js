import './styles/pokemonTable.css';
import React  from "react";
import PokemonItem from "./PokemonItem";


function PokemonTable(props){
    
    return (
        <table>
            <thead>
            </thead>
            <tbody>
                {props.listPokemon.map((value, index) => {
                    return (
                        <tr key={index}>
                            <PokemonItem data={value}></PokemonItem>
                        </tr>
                    )
                })}
            </tbody>
        </table>
    );
}

export default PokemonTable;