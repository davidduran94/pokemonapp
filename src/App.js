import logo from './logo.svg';
import './App.css';
import axios from 'axios';
import PokemonForm from './PokemonForm';
import PokemonTable from './PokemonTable';
import { useState } from 'react';


function App() {
  const [listPokemon, setListPokemon] = useState([]);

  const searchPokemon = (item) => {
    console.log("searching ...", item);
    
    axios.get(`https://pokeapi.co/api/v2/pokemon/${item}`)
      .then( (pokemon) => {
        reciveData(pokemon.data);
      })
      .catch((err) => {
        errorOnSearch(err);
      });
    
  };

  const errorOnSearch = err => {
    console.log("Pokemon not Found");
  }

  const reciveData = (data) => {
    console.log(data);
    let newPokemon = {};
    newPokemon.name = data.name;
    newPokemon.id = data.id;
    newPokemon.img = data.sprites.front_default;
    setListPokemon((listPokemon) => [...listPokemon, newPokemon]);
  };

  const handleSearch = (item) => {
    //buscar pokemon en API
    searchPokemon(item);
  };

  return (
    <div className="App">
      <header className="">
       <h1>Poke API</h1>
       <PokemonForm onNewPokemonSearch={handleSearch}></PokemonForm>
       {
         listPokemon.length == 0 ? <h3>No pokemon captured yet</h3> : <h3></h3>
       }
      </header>

      <PokemonTable listPokemon={listPokemon}></PokemonTable>
    </div>
  );
}

export default App;
