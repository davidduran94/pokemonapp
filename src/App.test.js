import { render, screen } from '@testing-library/react';
import { shallow, mount } from 'enzyme'
import App from './App';
import PokemonForm from './PokemonForm';
import PokemonTable from './PokemonTable';
import PokemonItem from './PokemonItem';
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const pokemon = {
  name: "pikachu",
  id: "25",
  img: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png",
};
const pokemons = [{
    name: "pikachu",
    id: "25",
    img: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png",
  },
  {
    name: "pikachu",
    id: "25",
    img: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png",
  },
  {
    name: "pikachu",
    id: "25",
    img: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png",
  }
];


test('renders without crash', () => {
  render(<App />);
});
test('renders without crash', () => {
  render(<PokemonForm />);
});

test('renders without crash', () => {
  render(<PokemonItem data={pokemon} />);
});
test('renders without crash', () => {
  render(<PokemonTable  listPokemon={pokemons}/>);
});

describe("Pokemon Item", () => {
  it("accepts pokemon props", () => {
    const wrapper = shallow(<PokemonItem data={pokemon} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Pokemon Table", () => {
  it("accepts pokemons props", () => {
    const wrapper = shallow(<PokemonTable listPokemon={pokemons} />);
    expect(wrapper).toMatchSnapshot();
  });
});

