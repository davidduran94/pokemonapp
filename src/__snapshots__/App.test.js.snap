// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Item accepts pokemon props 1`] = `ShallowWrapper {}`;

exports[`Pokemon Table accepts pokemon props 1`] = `ShallowWrapper {}`;

exports[`Pokemon Table accepts pokemons props 1`] = `ShallowWrapper {}`;
